utils/sendGridEventWebhook.js

```diff
@@ -0,0 +1,120 @@
+const sgMail = require("@sendgrid/mail");
+const { SecretManagerServiceClient } = require("@google-cloud/secret-manager");
+const { EventWebhook, EventWebhookHeader } = require("@sendgrid/eventwebhook");
+const { processEventWebhook } = require("./firestore");
+const { getResponseJSON } = require("./shared");
+
+const _testSendEmail = async (res) => {
+    console.log("Start sending email");
+
+    try {
+        // Add SendGrid apiKey to test
+        sgMail.setApiKey("");
+
+        // Update data to test
+        const msg = {
+            personalizations: [
+                {
+                    to: "",
+                    custom_args: {
+                        connect_id: "cnid1",
+                        token: "tk1",
+                        notification_id: "nid1",
+                        gcloud_project: process.env.GCLOUD_PROJECT,
+                        attempt: "1",
+                    },
+                },
+                // {
+                //     to: "",
+                //     custom_args: {
+                //         connect_id: "cnid2",
+                //         token: "tk2",
+                //         notification_id: "nid2",
+                //         gcloud_project: process.env.GCLOUD_PROJECT,
+                //         attempt: "1",
+                //     },
+                // },
+            ],
+            from: {
+                name:
+                    process.env.SG_FROM_NAME ||
+                    "Connect for Cancer Prevention Study",
+                email:
+                    process.env.SG_FROM_EMAIL ||
+                    "donotreply@myconnect.cancer.gov",
+            },
+            subject: "Test SendGrid Event Webhook",
+            content: [
+                {
+                    type: "text/html",
+                    value: "<p>Hello World!</p>",
+                },
+            ],
+            categories: ["Test"],
+        };
+        await sgMail.send(msg);
+        console.log("Complete sending email");
+        return res.status(200).json({ code: 200 });
+    } catch (e) {
+        console.error(e);
+        return res.status(500).json({ code: 500 });
+    }
+};
+
+const _receivedEvents = async (req, res) => {
```

Comment:
For my understanding, what is the meaning of an underscore before the variable name?

Reply:
I would like to mark it as an internal function used in the file (not export)